library/public/engine.h

```diff
@@ -0,0 +1,142 @@
+/**
+ * @class   engine
+ * @brief   Class used to create instance of other classes
+ *
+ * The main class of the libf3d to create all other instances
+ * Configured on creation using an enum, then all objects
+ * can be accessed through their getter.
+ *
+ * Example usage:
+ *
+ * \code{.cpp}
+ *  f3d::engine eng(f3d::engine::CREATE_WINDOW | f3d::engine::CREATE_INTERACTOR);
+ *  f3d::loader& load = eng.getLoader();
+ *  load.addFile("path/to/file");
+ *  load.loadFile(f3d::loader::LoadFileEnum::LOAD_CURRENT);
+ *  f3d::interactor& inter = eng.getInteractor();
+ *  inter.start();
+ * \endcode
+ */
+
+#ifndef f3d_engine_h
+#define f3d_engine_h
+
+#include "export.h"
+#include "window.h"
+
+#include <map>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace f3d
+{
+class options;
+class loader;
+class interactor;
+class F3D_EXPORT engine
+{
+public:
+  class exception : public std::runtime_error
```

Comment:
If there are multiple reasons for failure, consider making separate exceptions for each of them, so more granular `catch` statements are supported.

Reply:
good point

Reply:
Adressed in https://github.com/f3d-app/f3d/pull/355